### ISSUE TITLE
Android version...

### ISSUE THREAD

**Issue by intika** [intika/privafox-firefox#8]
Add android version

**Comment by Atavic** [comment]
> Also check Firefox lite

> Well it’s just like Firefox Focus with a better UI

See [here](https://www.ghacks.net/2017/10/02/mozilla-launches-firefox-rocket-browser-in-indonesia/)

For me Firefox Klar is worth a look. By unpacking firefox lite apk I see in /assets/licenses.html

> Adjust Android SDK

**Comment by Atavic** [comment]
> Also check Firefox lite

It’s Firefox Focus with a different UI for Indonesian users.

By unpacking the apk I see in /assets/licenses.html

> Adjust Android SDK

For me Firefox Klar is maybe worth a look.

**Comment by Atavic** [comment]
BTW even Klar has adjust partnership.

**Comment by Atavic** [comment]
Info: https://github.com/mozilla-mobile/focus-android/wiki/Adjust-Usage

**Comment by intika** [comment]
Thank you for the feedback i will look into that ;)

**Comment by intika** [comment]
I just checked the different versions they are nice... but i think i will stick with main Firefox for android for the moment... once all the rest will be done i will may be look at focus etc... i did added that to the check list in the first to see if i can adopt one of those instead of the main Firefox but it's not suitable as many important settings are removed from them like desktop version...

**Comment by elypter** [comment]
isnt firefox focus chrome based? it would be a completely different browser then

**Comment by intika** [comment]
it does a matter for the base the thing is the outcome which is have a fast reliable privacy aware browser for mobile; Andoird will need its own work and development any way.

**Comment by intika** [comment]
@elypter thank you for all your feedback and for participating actively in the project i will add you to the thanks you list... i also created an issue for that matter ;) https://github.com/intika/Librefox/issues/59

**Comment by Chippel** [comment]
So we should use Firefox focus on Android? Is it the same? I thought it is chrome based so it calls home I guess.

**Comment by intika** [comment]
If you want a privacy aware browser for mobile awaiting Librefox to provide an android version you can use https://www.bromite.org/

**Comment by intika** [comment]
Firefox to me is faster on mobile compared to chrome (less resources intensive) but when it come to privacy Bromite is way more better... the claims about privacy that are made by Mozilla on android are not clear.

**Comment by Chippel** [comment]
Do you have any proof that focus has less privacy or worse privacy than bromite?

**Comment by intika** [comment]
it depend on your needs and how you define privacy, this require a deep technical comparison, feel free to do so :) and choose the application that suit your needs ;)

**Comment by Atavic** [comment]
@elypter Due to the size, around 3 MB, it's most probably based on chrome.

**Comment by pashamcr** [comment]
> 
> 
> Do you have any proof that focus has less privacy or worse privacy than bromite?

I doubt that in the Play Market, you can find applications that respect privacy, only the presence of the application in the GP is suspicious. Don't be confused by the presence of:
_smali\com\adjust
smali\com\android\installreferrer
smali\com\google\android\finsky\externalreferrer_
in focus?,
smali\com\adjust
com\leanplum
com\google\firebase
smali\com\google\android\gms
in Firefox for Android?

check both browsers (bromite vs firefox/focus) with [it](https://f-droid.org/en/packages/com.oF2pks.classyshark3xodus/),
take the fingerprinting mitigations [test](https://www.bromite.org/detect), good results showed Firefox for Android along with [addons](https://github.com/intika/Librefox#recommended-addons),
also recommend: https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/
http://4pda.ru/forum/index.php?s=&showtopic=830910&view=findpost&p=64445735

**Comment by colbycdev** [comment]
Yes, an Android version is a must

**Comment by Atavic** [comment]
Both *ghacks-user.js* and *pyllyukko user.js* are lacking in Android reports, they have focused on the following solutions:

- **append the contents of your user.js file to prefs.js**

https://github.com/pyllyukko/user.js/issues/14#issuecomment-366571564

- **use remote debugging**

https://github.com/ghacksuserjs/ghacks-user.js/issues/318#issuecomment-359181951

Appending user.js is simpler, but who uses remote debugging could surely help...

[Get involved](https://github.com/ghacksuserjs/ghacks-user.js/issues/539), don't let these user.js repos and Librefox behind on Mobile!

**Comment by dimqua** [comment]
Firefox Focus (Klar) uses GeckoView engine. Firefox Lite is a fork, it uses WebView.

**Comment by fmilligan** [comment]
This is a must for Android users, like me. It looks and sounds great, but I can't actually use it

**Comment by intika** [comment]
>  It looks and sounds great, but I can't actually use it

@fmilligan can you please develop

**Comment by fmilligan** [comment]
@intika I'm not sure on the requirements, but if you give me what you need, I can certainly try

_Sent from my TETRA using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by intika** [comment]
When ready it will be released on the Playstore and F-droid...

**Comment by DI555** [comment]
Whould like to ask about one feature of it! If planned to develop fennec,- Android fennec version uses 2 different UI for tablet and smartphone types! Please add a switch/key to use that types manually!

**Comment by theel0ja** [comment]
https://f-droid.org/packages/org.mozilla.fennec_fdroid/

**Comment by theel0ja** [comment]
Firefox Focus, Klar, etc. aren't main Firefox based so they should just be forgotten.

https://www.ctrl.blog/entry/mozilla-gecko-focus

**Comment by androidacy-user** [comment]
As of now, without a full release, is it possible to apply the files to the Android version? Both rooted or unrooted? And can you link to extension used in librefox?
My goal is to attempt to get a librefox like experience. I tried bromite, and did not like it

**Comment by Atavic** [comment]
We only have the official Firefox for Android and Fennec left to review. Fennec seems a little more conservative - by looking at the about:config values - when compared to Firefox for Android.

**Comment by androidacy-user** [comment]
Isn't fennec also by Mozilla? 

_Sent from my TETRA using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Atavic** [comment]
Yes.

**Comment by Atavic** [comment]
...[Fennec F-Droid](https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/) is focused on removing any proprietary bits found in official Mozilla's builds.

**Comment by Atavic** [comment]
See: https://android.stackexchange.com/questions/194088/is-fennec-f-droid-an-official-product-by-mozilla

**Comment by dimqua** [comment]
Does Firefox for Android (and Fennec) have `mozilla.cfg` support?

**Comment by Atavic** [comment]
> As far as I know, Mozilla doesn't support enterprise deployments for Firefox for Android.

https://support.mozilla.org/en-US/questions/1202266

**Comment by bogachenkove** [comment]
@intika 

> Add android version (main firefox)

https://support.mozilla.org/ru/questions/1202266 Is it supported configuration by mozilla.cfg? No. This config is for computers only. ONLY. It is created for a company or organization to control office rats. Tell them which settings you can use and which ones you can't. If in the company one computer can be used by different people, the phone is an individual thing. Am I wrong? I have the wrong information?

On Android only user.js.
PS: think that and for applications "focus" also. so what are you going to add? JS-version?

**Comment by user858753257** [comment]
Hoe can I apply the enterprise desktop policies to the android version ?

**Comment by androidacy-user** [comment]
I do not believe you can. As far as I'm aware of, Android Firefox doesn't even have the required files in the first place
________________________________

**Comment by DI555** [comment]
@intika , could it be possible to build desktop ff version for android now?
(I didn't get it...... Isn't this topic about of? ;) )
...since mozilla imo plans to drop 'fennec' and targeting to 'focus' with geckoView(((

**Comment by bogachenkove** [comment]
>...since mozilla imo plans to drop 'fennec' and targeting to 'focus' with geckoView(((

???

**Comment by colbycdev** [comment]
@DI555 This issue is about the Android version, not the desktop version. 

_Sent from my TETRA using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by Serkan-devel** [comment]
why should the desktop version of firefox be ported to android. I'm shure mozilla still provides mobile firefox on the google playstore and I though F-Droid are the ones who create those firefox builds

**Comment by elypter** [comment]
the moble version doesnt offer nearly as many features?

**Comment by DI555** [comment]
Features and UI richness in descending:
desktop Firefox -> Fennec for Android -> Focus for Android
So, porting desktop Firefox to Android will be great leap forward for Android browsers!

**Comment by Serkan-devel** [comment]
@DI555 what are you talking about "features and UI descending"?
It's still the same codebase between desktop and mobile. I can still change the browser theme and install addons as usual
![screenshot_20190302-120205](https://user-images.githubusercontent.com/16866642/53681013-733b0000-3ce3-11e9-8d0a-320212b0794e.png)

**Comment by Serkan-devel** [comment]
I think what you suggest is reinventing the wheel basically

**Comment by ghc2018** [comment]
To get user.js support on firefox-android, wouldn't it be the best way to implement an addon for firefox? This way it would even be possible to "switch" to various user.js "profiles" which differ in only some special options e.g. to make several sites work even with a less relaxed user.js.
This site seems to explain how to create such an addon:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Implement_a_settings_page
